### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by RRRadicalEdward**
Hi @raptorswing! I'm going to use your crate for one of my pet projects. But, while exploring the source code I found that there were a lot of clippy warnings. So I fixed them :slightly_smiling_face:. After fixing, I ran unit tests and no one failed. 
Thank you for the project. I really like the code. 

Edit: I only left one warning untouchable:

As I don't know how to properly rename the module, or should I  even do it. Leaving it up to you.

### CODE REVIEW

File: src/dht/dht_settings.rs [hunk 1]
```diff
@@ -101,7 +101,13 @@ impl DHTSettings {
     }
 }
 
-#[derive(Clone)]
+impl Default for DHTSettings {
+    fn default() -> Self {
+        Self::default()
```
Comment (raptorswing) [inline src/dht/dht_settings.rs:106]: I think we should move the current implementation of `DHTSettings::default()` (lines 76-101) here, then delete `impl DHTSettings {...}` altogether.

```suggestion
        DHTSettings {
            token_secret_size: 10,
            max_peers_response: 128,
            max_sample_response: 50,
            min_sample_interval_secs: 10,
            router_ping_interval_secs: 900,
            reverify_interval_secs: 14 * 60,
            reverify_grace_period_secs: 15 * 60,
            verify_grace_period_secs: 60,
            get_peers_freshness_secs: 15 * 60,
            find_nodes_interval_secs: 33,
            find_nodes_skip_count: 32,
            max_torrents: 50,
            max_peers_per_torrent: 100,
            ping_check_interval_secs: 10,
            outgoing_request_prune_secs: 30,
            outgoing_reqiest_check_interval_secs: 30,
            read_only: false,
            routers: vec![
                "router.bittorrent.com:6881".to_string(),
                "router.utorrent.com:6881".to_string(),
                "dht.transmissionbt.com:6881".to_string(),
            ],
        }
```

File: src/packets/internal.rs [hunk 2]
```diff
@@ -87,22 +86,22 @@ pub enum DHTRequestSpecific {
...
     },
 
-    DHTFindNodeResponse {
+    FindNode {
         #[serde(rename = "r")]
         arguments: DHTFindNodeResponseArguments,
     },
 
-    DHTPingResponse {
+    PingResponse {
```
Comment (raptorswing) [inline src/packets/internal.rs:104]: Please remove `Response` from the end of this variant to be consistent with the others.

```suggestion
    Ping {
```

File: src/dht/socket.rs [hunk 3]
```diff
@@ -187,16 +187,13 @@ impl DHTSocket {
...
-                                }
-                            }
-                            None => { /*It's fine if there's no channel - just means that the sender didn't care about getting a response*/
+                        if let Some(response_channel) = request_info.response_channel {
+                            if let Err(e) = response_channel.send(message.clone()).await {
+                                let message = e.0;
+                                warn!(target: "rustydht_lib::DHTSocket", "Got response, but sending code abandoned the channel receiver. So sad. Response: {:?}. Sender: {:?}", message, sender);
                             }
                         }
+                        // It's fine if there's no channel - just means that the sender didn't care about getting a response
```
Comment (raptorswing) [inline src/dht/socket.rs:196]: Please move the comment to be before the `if let`
```suggestion
                        // It's fine if there's no channel - just means that the sender didn't care about getting a response
                        if let Some(response_channel) = request_info.response_channel {
                            if let Err(e) = response_channel.send(message.clone()).await {
                                let message = e.0;
                                warn!(target: "rustydht_lib::DHTSocket", "Got response, but sending code abandoned the channel receiver. So sad. Response: {:?}. Sender: {:?}", message, sender);
                            }
                        }
```

File: src/dht/dht.rs [hunk 4]
```diff
@@ -839,19 +838,13 @@ impl DHT {
...
-                                let mut state = state.lock().unwrap();
-                                for node in &args.nodes {
-                                    if node.id.is_valid_for_ip(&node.address.ip()) {
-                                        state.buckets.add_or_update(node.clone(), false);
-                                    }
+                        if let packets::ResponseSpecific::FindNodeResponse(args) = response_variant {
+                            let mut state = state.lock().unwrap();
+                            for node in &args.nodes {
+                                if node.id.is_valid_for_ip(&node.address.ip()) {
+                                    state.buckets.add_or_update(node.clone(), false);
```
Comment (raptorswing) [inline src/dht/dht.rs:845]: Please preserve the comment so I remember wtf I was thinking 😄 

```suggestion
                        // Special handling for find_node responses
                        // Add the nodes we got back as "seen" (even though we haven't necessarily seen them directly yet).
                        // They will be pinged later in an attempt to verify them.
                        if let packets::ResponseSpecific::FindNodeResponse(args) = response_variant {
                            let mut state = state.lock().unwrap();
                            for node in &args.nodes {
                                if node.id.is_valid_for_ip(&node.address.ip()) {
                                    state.buckets.add_or_update(node.clone(), false);
```

**Review by raptorswing — CHANGES_REQUESTED** [review]
Thanks very much for this PR. I can tell that you spent some time on it. I wasn't aware of `clippy` before - what a great tool. I'm incorporating it into my workflow from now on.

I have a few small requests for the changes - please take a look.

Thanks again.

File: src/dht/dht.rs (hunk 4 above)
Comment (RRRadicalEdward) [reply src/dht/dht.rs:848]: Done

File: src/dht/socket.rs (hunk 3 above)
Comment (RRRadicalEdward) [reply src/dht/socket.rs]: Done

File: src/dht/dht_settings.rs (hunk 1 above)
Comment (RRRadicalEdward) [reply src/dht/dht_settings.rs]: Done

File: src/packets/internal.rs (hunk 2 above)
Comment (RRRadicalEdward) [reply src/packets/internal.rs]: Done

**Review by raptorswing — APPROVED** [review]
Thanks!

**Pull request merged**